File: JoergIsAGeek.Workshop.Enterprise.WebApplication.ViewModels/MachineViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace JoergIsAGeek.Workshop.Enterprise.WebApplication.ViewModels {
  public class MachineViewModel : BaseViewModel {

    [Required]
    public string Name { get; set; }

    [StringLength(100)]
    public string Location { get; set; }

    public bool HasDevices { get; set; }
  }
}

File: JoergIsAGeek.Workshop.Enterprise.WebApplication/src/app/viewmodels/machine.viewmodel.ts
import { Display, Required } from 'svogv';

import { Base } from './base';

export class MachineViewModel extends Base {

  @Display('Machine Name', 10)
  @Required()
  name: string = '';

  @Display('Location', 20)
  @Required()
  location: string = '';


}
